core/src/main/java/com/linecorp/armeria/server/ProxiedAddresses.java

```diff
@@ -18,51 +18,86 @@
 
 import static java.util.Objects.requireNonNull;
 
-import java.net.SocketAddress;
+import java.net.InetSocketAddress;
+import java.util.List;
 
 import com.google.common.base.MoreObjects;
+import com.google.common.collect.ImmutableList;
 
 /**
  * An interface to provide source and destination addresses delivered from a proxy server.
  */
 public final class ProxiedAddresses {
 
     /**
-     * Creates a new instance.
+     * Creates a new instance with the specified {@code sourceAddress}.
      */
-    public static ProxiedAddresses of(SocketAddress sourceAddress, SocketAddress destinationAddress) {
-        return new ProxiedAddresses(sourceAddress, destinationAddress);
+    public static ProxiedAddresses of(InetSocketAddress sourceAddress) {
+        return new ProxiedAddresses(sourceAddress, ImmutableList.of());
     }
 
-    private final SocketAddress sourceAddress;
-    private final SocketAddress destinationAddress;
+    /**
+     * Creates a new instance with the specified {@code sourceAddress} and {@code destinationAddress}.
```

Comment:
Mention that port `0` means that the port number is unknown for all factory/getter Javadocs.